frontend/javascripts/oxalis/view/right-menu/trees_tab_view.js

```diff
@@ -181,13 +184,54 @@ export async function importTracingFiles(files: Array<File>, createGroupForEachF
       }
     };
 
+    const tryParsingFileAsZip = async file => {
+      try {
+        const zipFile = await JSZip().loadAsync(readFileAsArrayBuffer(file));
+        const nmlFileName = Object.keys(zipFile.files).find(
+          key => _.last(key.split(".")) === "nml",
+        );
+        const nmlFile = await zipFile.file(nmlFileName).async("blob");
+        const nmlImportActions = await tryParsingFileAsNml(nmlFile);
+
+        const dataFileName = Object.keys(zipFile.files).find(
+          key => _.last(key.split(".")) === "zip",
+        );
+        if (dataFileName) {
+          const dataBlob = await zipFile.file(dataFileName).async("blob");
+          const dataFile = new File([dataBlob], dataFileName);
+
+          const { tracing, dataset } = Store.getState();
+          const newLargestSegmentId = await importTracing(tracing, dataFile);
+
+          if (tracing && tracing.volume) {
+            Store.dispatch(importTracingAction()); // TODO put all in one Action, which gets consumed twice (volume tracing reducer + undo save saga)
+            Store.dispatch(
+              setVersionNumberAction(tracing.volume ? tracing.volume.version + 1 : 1, "volume"),
+            );
+            Store.dispatch(setMaxCellAction(newLargestSegmentId));
+            await clearCache(dataset, tracing.volume ? tracing.volume.tracingId : "whyFlow");
+            api.data.reloadBuckets(tracing.volume ? tracing.volume.tracingId : "whyFlow");
+            window.needsRerender = true;
+          }
+        }
+
+        return nmlImportActions;
+      } catch (error) {
+        console.error(`Tried parsing file "${file.name}" as ZIP but failed. ${error.message}`);
+        return undefined;
+      }
+    };
+
     const { successes: importActionsWithDatasetNames, errors } = await Utils.promiseAllWithErrors(
       files.map(async file => {
         const ext = _.last(file.name.split("."));
-        const tryImportFunctions =
-          ext === "nml" || ext === "xml"
-            ? [tryParsingFileAsNml, tryParsingFileAsProtobuf]
-            : [tryParsingFileAsProtobuf, tryParsingFileAsNml];
+        let tryImportFunctions;
+        if (ext === "nml" || ext === "xml")
+          tryImportFunctions = [tryParsingFileAsNml, tryParsingFileAsProtobuf, tryParsingFileAsZip];
+        else if (ext === "zip")
+          tryImportFunctions = [tryParsingFileAsZip, tryParsingFileAsNml, tryParsingFileAsProtobuf];
```

Comment:
I know the old code already had this "let's try everything" approach, but I think for the ZIP vs not-ZIP discrimination it's not very reasonable. I suggest to not run `tryParsingFileAsZip` if the file is nml/xml and to only run `tryParsingFileAsZip` if it's a zip.